tests/commands/test_get_nonce.py

```diff
@@ -1,55 +1,47 @@
 """Tests for get-nonce command."""
 import logging
-import os
-from unittest.mock import patch
+from unittest.mock import AsyncMock, patch
 
 import pytest
 
-from nile.utils.get_nonce import get_nonce
+from nile.common import set_args
+from nile.utils.get_nonce import get_nonce, get_nonce_without_log
 
-GATEWAYS = {"localhost": "http://127.0.0.1:5050/"}
+NONCE = 5
 
 
+@pytest.mark.asyncio
 @pytest.mark.parametrize(
     "contract_address, network",
     [("0xffff", "localhost"), ("0xffff", "goerli"), ("0xffff", "mainnet")],
 )
-@patch("nile.core.node.subprocess.check_output")
-@patch("nile.common.get_gateways", return_value=GATEWAYS)
-def test_call_format(mock_gateway, mock_subprocess, contract_address, network):
-    get_nonce(contract_address, network)
-
-    command = ["starknet", "get_nonce", "--contract_address", contract_address]
-    if network == "localhost":
-        command.append("--feeder_gateway_url=http://127.0.0.1:5050/")
-    else:
-        assert os.getenv("STARKNET_NETWORK") == f"alpha-{network}"
-
-    mock_subprocess.assert_called_once_with(command)
+@patch("nile.utils.get_nonce.get_nonce_without_log")
+async def test_get_nonce(mock_without_log, contract_address, network, caplog):
+    logging.getLogger().setLevel(logging.INFO)
 
+    mock_without_log.return_value = NONCE
 
-@patch("nile.core.node.subprocess.check_output", return_value="5")
-@patch("nile.common.get_gateways", return_value=GATEWAYS)
-def test_get_nonce(mock_gateway, mock_subprocess, caplog):
-    logging.getLogger().setLevel(logging.INFO)
+    # Check return value
+    nonce = await get_nonce(contract_address, network)
+    assert nonce == NONCE
 
-    # check return values
-    nonce = get_nonce("0xffff", "localhost")
-    assert nonce == 5
+    # Check internal
+    mock_without_log.assert_called_once_with(contract_address, network)
```

Comment:
does it make sense to test that `get_nonce` calls an internal function? it seems like testing implementation details. any reason not to have a single test of just behavior as the one below?

Reply:
I think it makes sense. It's eliminating any opportunity for the internal func to manipulate the test. It's probably not a big deal in this simple scenario. Happy to change it if preferred

Reply:
yes, i wouldn't test an auxiliary function call. that's an implementation detail.